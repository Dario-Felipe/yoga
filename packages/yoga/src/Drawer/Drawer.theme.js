const Drawer = ({ spacing }) => ({
  padding: {
    top: spacing.small,
    right: spacing.xxxlarge,
    bottom: spacing.xxlarge,
    left: spacing.xxxlarge,
  },
  width: {
    default: 600,
  },
});

export default Drawer;
