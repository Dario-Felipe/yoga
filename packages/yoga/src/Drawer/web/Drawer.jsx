import React from 'react';
import styled from 'styled-components';
import { node, string } from 'prop-types';
import Dialog from '../../Dialog';

const StyledDrawer = styled(Dialog)`
  ${({
    theme: {
      yoga: {
        components: { drawer },
      },
    },
  }) => `
  padding: ${drawer.padding.top}px ${drawer.padding.right}px ${drawer.padding.bottom}px ${drawer.padding.left}px;
  width: ${drawer.width.default}px;
  `}
  border-radius: 0!important;
  height: 100%;
  align-self: flex-end;
  position: absolute;
  right: 0;
  animation: content;
  animation-duration: 400ms;
  animation-fill-mode: forwards;
  transition: 0.25s ease-in-out;
  @keyframes content {
    0% {
      transform: translate3d(100%, 0, 0);
    }
  }
`;

function Drawer({ testId, ...props }) {
  return <StyledDrawer data-testid={testId} {...props} />;
}

Drawer.propTypes = {
  children: node.isRequired,
  testId: string,
};

Drawer.defaultProps = {
  testId: undefined,
};

export default Drawer;
