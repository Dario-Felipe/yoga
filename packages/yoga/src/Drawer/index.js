import Drawer from './web';

export default Drawer;
