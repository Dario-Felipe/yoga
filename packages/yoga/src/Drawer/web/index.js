import Drawer from './Drawer';
import Header from './Header';

Drawer.Header = Header;

export default Drawer;
