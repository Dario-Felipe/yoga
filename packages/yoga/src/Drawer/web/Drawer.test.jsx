import React from 'react';
import { screen, render, fireEvent, cleanup } from '@testing-library/react';

import { ThemeProvider, Button } from '../..';

import Drawer from '.';

describe('<Drawer />', () => {
  afterEach(cleanup);

  it('should match snapshot', () => {
    const { baseElement } = render(
      <ThemeProvider>
        <Drawer isOpen>
          <Drawer.Header>Title</Drawer.Header>
          <Drawer.Content>Subtitle</Drawer.Content>
          <Drawer.Footer>
            <Button secondary>Ok, got it</Button>
          </Drawer.Footer>
        </Drawer>
      </ThemeProvider>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('should render a minimal drawer', () => {
    const onActionMock = jest.fn();

    render(
      <ThemeProvider>
        <Drawer isOpen>
          <Drawer.Header>Title</Drawer.Header>
          <Drawer.Content>Subtitle</Drawer.Content>
          <Drawer.Footer>
            <Button onClick={onActionMock} secondary>
              Ok, got it
            </Button>
          </Drawer.Footer>
        </Drawer>
      </ThemeProvider>,
    );

    screen.getByText('Title');
    screen.getByText('Subtitle');

    const button = screen.getByRole('button', { name: /Ok, got it/i });

    fireEvent.click(button);

    expect(onActionMock).toHaveBeenCalledTimes(1);
  });
});
