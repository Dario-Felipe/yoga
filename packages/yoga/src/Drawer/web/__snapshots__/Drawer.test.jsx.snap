// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Drawer /> should match snapshot 1`] = `
.c8 {
  box-sizing: border-box;
  text-align: center;
  outline: none;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: auto;
  height: 48px;
  padding-left: 24px;
  padding-right: 24px;
  background-color: #231B22;
  border: none;
  border-radius: 9999px;
  color: #FFFFFF;
  font-size: 16px;
  font-weight: 500;
  font-family: Rubik;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: 1;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c8 svg {
  width: 24px;
  height: 24px;
  fill: #FFFFFF;
  margin-right: 8px;
  -webkit-transition: fill 0.2s;
  transition: fill 0.2s;
}

.c8:not([disabled]):hover,
.c8:not([disabled]):focus {
  box-shadow: 0 4px 8px rgba(35,27,34,0.45);
}

.c8:active {
  background-color: rgba(35,27,34,0.75);
  color: #FFFFFF;
}

.c8:active svg {
  fill: #FFFFFF;
}

.c4 {
  width: 100%;
  color: #231B22;
}

.c5 {
  margin-bottom: 32px;
  color: #6B6B78;
  font-size: 16px;
  font-weight: 400;
  text-align: center;
}

.c7 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  padding: 16px  16px  16px  16px;
  border-radius: 8px;
  background-color: #FFFFFF;
  box-shadow: 0 2px 4px -1px rgba(152,152,166,0.2),0 4px 5px 0px rgba(152,152,166,0.14),0 1px 10px 0px rgba(152,152,166,0.12);
}

.c2 {
  padding: 40px  32px  32px;
  width: 580px;
  min-height: 160px;
  border-radius: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  z-index: 3;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  bottom: 0;
  background-color: rgba(35,27,34,0.48);
}

.c3 {
  padding: 16px 56px 40px 56px;
  width: 600px;
  border-radius: 0!important;
  height: 100%;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  position: absolute;
  right: 0;
  -webkit-animation: content;
  animation: content;
  -webkit-animation-duration: 400ms;
  animation-duration: 400ms;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  -webkit-transition: 0.25s ease-in-out;
  transition: 0.25s ease-in-out;
}

.c6 {
  margin: 0;
  height: 100%;
  width: 100%;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  text-align: left;
}

<body>
  <div />
  <div
    id="dialog"
  >
    <div>
      <div
        class="c0"
      >
        <section
          class="c1 c2 c3"
        >
          <header
            class="c4"
            color="text.primary"
            width="100%"
          >
            Title
          </header>
          <div
            class="c5 c6"
            color="text.secondary"
          >
            Subtitle
          </div>
          <footer
            class="c7"
            d="flex"
            width="100%"
          >
            <button
              class="c8"
            >
              Ok, got it
            </button>
          </footer>
        </section>
      </div>
    </div>
  </div>
</body>
`;
